src/scss/style.scss

```diff
@@ -599,6 +600,18 @@ button {
   -webkit-box-sizing: border-box;
   padding: 1px;
   text-align: center;
+  transition: width 100ms ease-out;
+  &:focus {
+    position: absolute;
+    left: 50%;
+    top: 50%;
+    transform: translate(-50%, -7px);
+    z-index: 2;
+    transition: width 0ms linear;
```

Comment:
The nitpick.

```suggestion
    transition: width 100ms linear;
```

Reply:
Hi @noroadsleft 

The only downside (which is why I made it 0ms originally) is as you type the width changes 100ms after which might not be a desirable effect. To sort this you would need to have a "focus/blur" event and a"typing" event. Did you want me to take a look at that?

Reply:
Yeah, I would prefer that. The thing about the instant swap is that sometimes it comes off looking like a CSS glitch (not a criticism of your changes, but of design as a whole), whereas the animation – however short or long – clarifies that it's intended behaviour.

Reply:
christopher, you could also use transition tag, and just make it animate in there to 100% then keep the widening thing for typing only.

Reply:
@yanfali Yeah I would need to implement an event in js for typing on vs the focus/blur event and this would fix it.

@noroadsleft Could do that as well, its 100ms though so would be quite a sharp animation, pending the user and how fast they type it might not be the desired effect

Definitely not precious on the approach and this is your baby, happy to take direction from you guys on what you would what approach you would like to take (segue into saying thanks for the effort you and other contributors have put into the site so far, made it easy to update my keyboard)